shelljs/shelljs.d.ts

```diff
@@ -434,31 +436,44 @@ declare module "shelljs"
      * Object containing environment variables (both getter and setter). Shortcut to process.env.
      */
     export var env: { [key: string]: string };
-
-    /*
-
-    // Not yet implemented due to implementation issues (constant overloads and return types).
-    // See: https://github.com/arturadib/shelljs#execcommand--options--callback
-
-    export function exec(command: string, options: ExecOptions, callback: (code: number, output: string) => any): any;
-    export function exec(command: string, options: ExecOptions): any;
-
-    interface ExecOptions
-    {
-        silent: boolean;
-        async: boolean;
-    }
-
-    */
-
+    
     /**
      * Executes the given command synchronously.
-     * @param  {string}          command The commadn to execute.
+     * @param  {string}          command The command to execute.
      * @return {ExecReturnValue}         Returns an object containing the return code and output as string.
      */
-    export function exec(command: string): ExecReturnValue;
+    export function exec(command: string): ExecOutputReturnValue;
+    /**
+     * Executes the given command synchronously.
+     * @param  {string}          command The command to execute.
+     * @param  {ExecOptions}     options Silence and synchronous options.
+     * @return {ExecReturnValue}         Returns an object containing the return code and output as string.
+     */
+    export function exec(command: string, options: ExecOptions): ExecOutputReturnValue;
```

Comment:
Well, correct me if I'm wrong, but the [docs](https://nodejs.org/api/child_process.html#child_process_class_childprocess) say that `ChildProcess` actually is a class. `ChildProcess` seems to be defined as an interface in node.d.ts.

I'm okay with both, `any` as a return type or a union return. Because the caller has to cast anyway, it wouldn't make a big difference. A union would give the caller (and the compiler) a hint, so I'd prefer that, if that's possible.

If this little issue is fixed (that is having either any or a union as a return type for this overload), I'm fine with this PR.